doc/architecture/squid.md

```diff
@@ -559,25 +476,213 @@ const success = ocean.accounts.requestTokens(account, amount)
 
 ---
 
+## ocean.compute
+
+### order
+
+Starts an order of a compute service that is defined in an asset's services.
+
+_Parameters_
+
+```
+         datasetDid: str -- The DID of the asset (of type `dataset`) to run the algorithm on.
+    consumerAccount: `Account` instance -- The account of the consumer ordering the service (optional, depends on Web3 implementation)
+```
+
+_Returns_
+
+```
+agreementId: hex str -- The service agreement ID, representation of `bytes32` ID.
+
+``` 
+
+_Example_
+
+```js
+const agreementId = ocean.compute.order(did, consumerAccount)
+```
+
+---
+
+### start
+
+Executes a compute job in a remote compute service associated with an asset and a service agreement ID.
+
+_Parameters_
+
+```
+        agreementId: hex str -- The service agreement ID, representation of `bytes32` ID.
+         datasetDid: str -- The DID of the asset (of type `dataset`) to run the algorithm on.
+    consumerAccount: `Account` instance -- The account of the consumer ordering the service.
+       algorithmDid: str -- The DID of the algorithm asset (of type `algorithm`) to run on the asset.
+       algorithmRaw: str -- The raw text of the algorithm to run in the compute job (e.g. a jupyter notebook) or a valid URL to fetch the algorithm.
+      algorithmMeta: `AlgorithmMetadata` instance -- Metadata about the algorithm being run if `algorithm` is being used. This is ignored when `algorithmDid` is specified.
+```
+
+_Returns_
+
+```
+str -- The ID of the compute job.
+``` 
+
+_Example_
+
+```js
+const jobId = ocean.compute.start(agreementId, did, consumerAccount, algorithmDid)
+```
+
+### status
+
+Returns information about the status of a compute job
+
+_Parameters_
+
+```
+agreementId: str -- The ID of the service agreement.
+jobId: str -- The ID of the job.
+```
+
+_Returns_
```

Comment:
the status responses should be defined further

Reply:
Agreed.

Reply:
defined based on Brizo API status responses https://github.com/oceanprotocol/brizo/blob/feature/compute/API.md